test/e2e_test.go

```diff
@@ -108,6 +120,8 @@ var _ = Describe("MetricsServer", func() {
 		panic(err)
 	}
 
+	needTestSideCarsContainers = getSidecarFeature(client)
```

Comment:
two naming suggestions here:
- `getSidecarFeature` should be renamed to `hasSidecarFeature` or `hasSidecarFeatureEnabled` since it returns a bool and not the actual feature object
- `needTestSideCarsContainers` could be renamed to `testSideCarsContainers`